tsweb/tsweb.go

```diff
@@ -177,29 +185,14 @@ type ReturnHandler interface {
 	ServeHTTPReturn(http.ResponseWriter, *http.Request) error
 }
 
-// BucketedStatsOptions describes tsweb handler options surrounding
-// the generation of metrics, grouped into buckets.
-type BucketedStatsOptions struct {
-	// Bucket returns which bucket the given request is in.
-	// If nil, [NormalizedPath] is used to compute the bucket.
-	Bucket func(req *http.Request) string
-
-	// If non-nil, Started maintains a counter of all requests which
-	// have begun processing.
-	Started *metrics.LabelMap
-
-	// If non-nil, Finished maintains a counter of all requests which
-	// have finished processing (that is, the HTTP handler has returned).
-	Finished *metrics.LabelMap
-}
-
 // normalizePathRegex matches components in a HTTP request path
 // that should be replaced.
 //
 // See: https://regex101.com/r/WIfpaR/3 for the explainer and test cases.
 var normalizePathRegex = regexp.MustCompile("([a-fA-F0-9]{9,}|([^\\/])+\\.([^\\/]){2,}|((n|k|u|L|t|S)[a-zA-Z0-9]{5,}(CNTRL|Djz1H|LV5CY|mxgaY|jNy1b))|(([^\\/])+\\@passkey))")
```

Comment:
I'm a bit unsure if it makes sense to try to solve it in a generic way here, and if it might be better as a whole to move this to the caller/creator of the handler to override instead of chaining it here and making it ever so slightly unpredictable on the other side of the codebase.